src/Processors/Formats/Impl/SQLiteInputFormat.cpp

```diff
@@ -0,0 +1,161 @@
+    const RowInputFormatParams & params_,
+    const FormatSettings & format_settings_)
+    : IRowInputFormat(header_, in_, params_), table_name(format_settings_.sqlite.table_name), format_settings(format_settings_)
+{
+    initMemVFS();
+}
+
+void SQLiteInputFormat::prepareReader() {
+    std::atomic<int> is_stopped = 0;
+    file_reader = asArrowFile(*in, format_settings, is_stopped, "sqlite3", "");
+    //file_reader = asArrowFile(*in, format_settings, is_stopped, "sqlite3", "SQLite format 3");
+
+    std::ostringstream ss;
+	ss << file_reader.get();
+	std::string uri = ss.str();
+
+    sqlite3 * db_ptr = nullptr;
+    int status = sqlite3_open_v2(uri.c_str(), &db_ptr, SQLITE_OPEN_READONLY | SQLITE_OPEN_URI, "ch_read_vfs");
+    if (status != SQLITE_OK) {
+        throw Exception::createDeprecated(fmt::format("Cannot open sqlite database. Error status: {}. Message: {}",
+                                       status, sqlite3_errstr(status)), ErrorCodes::SQLITE_ENGINE_ERROR);
+    }
+    db.reset(db_ptr, sqlite3_close_v2); 
+}
+
+std::vector<String> SQLiteInputFormat::getTablesNames() {
+    std::vector<String> tables_names;
+    std::string query = "SELECT name FROM sqlite_master "
+                        "WHERE type = 'table' AND name NOT LIKE 'sqlite_%'";
+    
+    auto callback_get_data = [](void * res, int col_num, char ** data_by_col, char ** /* col_names */) -> int
+    {   
+        for (int i = 0; i < col_num; ++i)
+            static_cast<std::vector<String> *>(res)->push_back(String(data_by_col[i]));
+        return 0;
+    };
+
+    
+    
+    int status = sqlite3_exec(db.get(), query.c_str(), callback_get_data, &tables_names, nullptr);
+    
+    if (status != SQLITE_OK)
+    {
+        throw Exception::createDeprecated(fmt::format("Failed to fetch SQLite tables names. Error status: {}. Message: {}",
+                                       status, sqlite3_errstr(status)), ErrorCodes::SQLITE_ENGINE_ERROR);
+    }
+    
+    return tables_names;
+}
+
+void SQLiteInputFormat::readPrefix()
+{
+    prepareReader();
+
+    auto tables_names = getTablesNames();
+    if (table_name.empty()) {
+        if (tables_names.empty()) {
+            throw Exception(ErrorCodes::UNKNOWN_TABLE, "Can't get any table");
+        }
+        table_name = tables_names[0];
+    } else {
+        if (std::find(tables_names.begin(), tables_names.end(), table_name) == tables_names.end()) {
+            throw Exception::createDeprecated(fmt::format("Failed to find SQLite {} table", table_name), ErrorCodes::UNKNOWN_TABLE);
+        }
+    }
+    
+    /*
+    auto names_and_types = fetchSQLiteTableStructure(db.get(), table_name);
+    
+    serializations.resize(names_and_types->getTypes().size());
+    for (size_t i = 0; i < serializations.size(); i++) {
+        serializations[i] = names_and_types->getTypes()[i]->getDefaultSerialization();
+    }
+    */
+
+    std::string select_query = fmt::format("SELECT * FROM {};", table_name);
```

Comment:
I wonder if we can reuse `SQLiteSource` (or its parts) here instead of querying and deserializing by hand. It has more features:
 * It selects only the requested columns.
 * It applies filters from WHERE when possible. (Currently this is not possible with input formats, but https://github.com/ClickHouse/ClickHouse/pull/52951 makes it possible.)
 * It reads results in binary form rather than parsing from text.